AY2122S2-CS2103T-T13-3/tp
# Show uncompleted meetings

## Related issues
Fixes #74.

## Description


Enforces the filtered meeting list to always only show meetings that are not yet completed (e.g. recurring, ongoing, and upcoming meetings).

## How to test

1. Add a non-recurring meeting that will expire in a short moment (e.g. `startDateTime` + `duration` will expire in two minutes from now).
2. After the short moment has elapsed, run any command (e.g. `list`). The completed meeting should now be absent from the GUI.

## Other information

This implementation introduces an invariant for the filtered meeting list; the filtered meeting list will filter out any completed meetings after every command. Since this invariant is dependent on the current date and time, current test cases are failing. This would need to be addressed somehow.

This implementation also allows the invariant to be negated; the filtered meeting list will filter out any uncompleted meetings after every command. This can be achieved through `ModelManager::showCompletedMeetings`, and can be used to implement the archive command as well.

While I believe implementing this feature as an invariant makes sense, it'll be great to get some alternate viewpoints as well. This is especially important as this approach would modify the behaviour of something fundamental like `ModelManager`.
